lib/net/ssh/multi/session.rb

```diff
@@ -447,7 +450,7 @@ def io_select_wait(wait)
 
     def keepalive_interval
       servers = server_list.select { |s| s.options[:keepalive] }
-      servers.map { |s| s.options[:keepalive_interval] }.compact.min
+      servers.map { |s| s.options[:keepalive_interval] || DEFAULT_IO_SELECT_TIMEOUT }.compact.min
```

Comment:
I don't think your proposal solves DEFAULT_IO_SELECT_TIMEOUT . I don't think that net-ssh-multi should be concerned with timeout. It should ask the ssh session of how soon it need to be run pre/postprocess callbacks. So i think something like ```session.max_select_wait_time```  would work better.
Then max_selcect_wait = server_list.map(&:max_select_wait_time).min